odoo/addons/base/tests/common.py

```diff
@@ -43,40 +98,80 @@ def _enable_currency(cls, currency_code):
         return currency
 
     @classmethod
-    def _use_currency(cls, currency_code):
+    def _use_currency(cls, company, currency_code):
         # Enforce constant currency
         currency = cls._enable_currency(currency_code)
-        if not cls.env.company.currency_id == currency:
-            cls.env.transaction.cache.set(cls.env.company, type(cls.env.company).currency_id, currency.id, dirty=True)
+        if company.currency_id != currency:
+            cls.env.transaction.cache.set(company, type(company).currency_id, currency.id, dirty=True)
             # this is equivalent to cls.env.company.currency_id = currency but without triggering buisness code checks.
             # The value is added in cache, and the cache value is set as dirty so that that
             # the value will be written to the database on next flush.
             # this was needed because some journal entries may exist when running tests, especially l10n demo data.
 
+    @classmethod
+    def _create_partner(cls, **create_values):
+        return cls.env['res.partner'].create({
+            'name': "Test Partner",
+            'company_id': False,
+            **create_values,
+        })
+
+    @classmethod
+    def _create_user(cls, groups, **create_values):
+        return cls.env['res.users'].create({
+            'name': "Test User",
+            'login': 'test_user',
+            'password': 'test_user',
```

Comment:
Don't give password unless you want to test something related to passwords: it takes a lot of time just to compute hashes and perform checks (and by a lot, it was really a lot). Better skip password when possible (aka most tests probably).